src/sagemaker/local/image.py

```diff
@@ -490,6 +484,18 @@ def _create_config_file_directories(root, host):
         os.makedirs(os.path.join(root, host, d))
 
 
+def _delete_tree(path):
+    try:
+        shutil.rmtree(path)
+    except OSError as exc:
+        # handle permission denied gracefully. Any other error we will raise the exception up.
```

Comment:
Put a short explanation of the docker writing-files-as-root issue here in a comment.